crates/biome_js_analyze/tests/specs/nursery/noUselessUndefinedInitialization/invalid.js.snap

```diff
@@ -0,0 +1,218 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalid.js
+---
+# Input
+```jsx
+const inline1 = undefined,
+	inline2 = undefined;
+let test1,
+	test2 = undefined;
+var c = undefined,
+	o = undefined;
+
+for (let i = 0; i < 100; i++) {
+	let i = undefined;
+}
+
+let x = 1, y = undefined, z = 40
+
+let /* comment */d = undefined;
+
+let e = undefined/**/ ;
+let f = /**/undefined/**/ ;
+let g 
+  /**/= /**/undefined/**/ ;
+```
+
+# Diagnostics
+```
+invalid.js:4:2 lint/nursery/noUselessUndefinedInitialization  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! It's not necessary to initialize test2 to undefined.
+  
+    2 │ 	inline2 = undefined;
+    3 │ let test1,
+  > 4 │ 	test2 = undefined;
```

Comment:
The diagnostic is highlighting the whole assignment, but it's incorrect. The diagnostic should only align starting from `=`.

You can use the state of the rule to save the `TextRange` of the right hand side of the assignment

Reply:
I replicated Eslint's behaviour on this matter ([Eslint's Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW5kZWYtaW5pdDogXCJlcnJvclwiKi9cblxudmFyIGZvbyA9IC8qKi91bmRlZmluZWQvKiovO1xubGV0IGJhciA9IHVuZGVmaW5lZDtcblxubGV0IGlubGluZSxcbiAgaW5saW5lMiA9IHVuZGVmaW5lZDsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)).

If you think we should diverge on this too, I'll gladly change the logic for the diagnostic!

Reply:
We don't usually check their diagnostics, our "parity" is around the logic of the rule (what to trigger and what to not trigger).

 Code fixes and diagnostics are our own :)

Reply:
Oh okay, thanks!
I'll change it in order to make the diagnostic start from the `=` token!